query/src/test/java/com/connexta/replication/query/cmp/QueryServiceComponentTest.java

```diff
@@ -0,0 +1,421 @@
+import org.springframework.test.context.junit4.SpringRunner;
+
+@ComponentScan(
+    basePackageClasses = {
+      ComponentTestConfig.class,
+      FilterIndexRepository.class,
+      EmbeddedSolrConfig.class
+    })
+@EnableSolrRepositories(basePackages = "com.connexta.replication")
+@RunWith(SpringRunner.class)
+public class QueryServiceComponentTest {
+  private static final int QUEUE_CAPACITY = 3;
+  private static final long THIRTY_MINUTES = TimeUnit.MINUTES.toSeconds(30L);
+  private static final MeterRegistry REGISTRY = new MeterRegistryMock();
+
+  private final QueueBroker broker =
+      new MemoryQueueBroker(
+          QueryServiceComponentTest.QUEUE_CAPACITY, QueryServiceComponentTest.REGISTRY);
+
+  @Autowired private FilterIndexManager indexManager;
+  @Autowired private FilterManager filterManager;
+  @Autowired private SiteManager siteManager;
+
+  @Rule public final MultiThreadedErrorCollector collector = new MultiThreadedErrorCollector();
+
+  private NodeAdapters nodeAdapters = mock(NodeAdapters.class);
+  private NodeAdapterFactory adapterFactory = mock(NodeAdapterFactory.class);
+  private final Set<Task> tasks = new HashSet<>();
+  private final NodeAdapter adapter = mock(NodeAdapter.class);
+
+  @Before
+  public void setup() {
+    when(adapterFactory.create(any(URL.class))).thenReturn(adapter);
+    when(nodeAdapters.factoryFor(SiteType.DDF)).thenReturn(adapterFactory);
+    when(nodeAdapters.factoryFor(SiteType.ION)).thenAnswer(collectAdapterException());
+    when(adapter.isAvailable()).thenReturn(true);
+  }
+
+  private Answer<?> collectAdapterException() {
+    return (Answer<Object>)
+        invocationOnMock -> {
+          Throwable t = new AdapterException("no adapters for ION sites should be retrieved");
+          collector.addError(t);
+          throw t;
+        };
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    tasks.clear();
+    filterManager.objects().map(Filter::getId).forEach(filterManager::remove);
+    siteManager.objects().map(Site::getId).forEach(siteManager::remove);
+  }
+
+  /*
+   Verifies that multiple metacards can be handled and that the proper date is saved in the filterIndex
+  */
+  @Test
+  public void testQueryMultipleMetacards() throws Exception {
+    final Site site = new SiteImpl(generateSitePojo(1));
+    siteManager.save(site);
+    final Filter filter = new FilterImpl(generateFilterPojo("1", 11));
+    filterManager.save(filter);
+    List<Metadata> metadataList =
+        List.of(
+            generateMetadataWithResource(111),
+            generateMetadataWithResource(112),
+            generateMetadataWithResource(113, new Date(1)));
+    when(adapter.query(Mockito.argThat(isQueryRequestWithFilter(filter.getFilter()))))
+        .thenReturn(
+            new QueryResponseImpl(
+                metadataList,
+                e -> {
+                  throw new AssertionError(e);
+                }));
+
+    indexManager = Mockito.spy(indexManager);
```

Comment:
❓ Wouldn't you need to pass the spied index manager to your tools in order for it to use the spy and you being able to verify calls on it?